src/wallet/db.cpp

```diff
@@ -52,6 +53,24 @@ void CheckUniqueFileid(const CDBEnv& env, const std::string& filename, Db& db)
         }
     }
 }
+
+bool LockEnvDirectory(const fs::path& env_path)
+{
+    // Make sure only a single Bitcoin process is using the wallet directory.
+    fs::path lock_file_path = env_path / ".lock";
```

Comment:
What if the walletdir is the same as the datadir? Won't there be a problem if both use the same lockfile? I would consider renaming this to `".walletlock"` perhaps (which is also clearer to people who see the file).

Reply:
Reading more, I think this is indeed advisable:

From http://www.boost.org/doc/libs/1_55_0/doc/html/interprocess/synchronization_mechanisms.html#interprocess.synchronization_mechanisms.file_lock:
> * It's unspecified if a file_lock synchronizes two threads from the same process.
> * It's unspecified if a process can use two file_lock objects pointing to the same file.

Reply:
Good point, fixed!